src/main/java/org/fcrepo/camel/FcrepoPrefer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.camel;
+
+import static java.util.Arrays.stream;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A class representing the value of an HTTP Prefer header
+ *
+ * @author Aaron Coburn
+ */
+public class FcrepoPrefer {
+
+    private static final String LINK_DELIM = ";\\s*";
+
+    private static final String OMIT = "omit";
+
+    private static final String INCLUDE = "include";
+
+    private String returnType = null;
+
+    private List<URI> omit;
+
+    private List<URI> include;
+
+    /**
+     * Create a representation of a Prefer header.
+     *
+     * @param prefer the value for a Prefer header
+     */
+    public FcrepoPrefer(final String prefer) {
+        parse(prefer);
+    }
+
+    /**
+     * Whether a minimal representation has been requested
+     * @return whether this is a minimal request
+     */
+    public boolean isMinimal() {
+        return returnType != null && returnType.equals("minimal");
+    }
+
+    /**
+     * Whether a non-minimal representation has been requested
+     * @return whether this is a non-minimal request
+     */
+    public boolean isRepresentation() {
+        return returnType != null && returnType.equals("representation");
+    }
+
+    /**
+     * Retrieve the omit portion of a Prefer header
+     *
+     * @return the omit portion of a Prefer header
+     */
+    public List<URI> getOmit() {
+        return omit;
+    }
+
+    /**
+     * Retrieve the include portion of the prefer header
+     *
+     * @return the include portion of a Prefer header
+     */
+    public List<URI> getInclude() {
+        return include;
+    }
+
+    /**
+     * Parse the value of a prefer header
+     */
+    private void parse(final String prefer) {
+        if (prefer != null) {
+            final Map<String, String> data = new HashMap<>();
+
+            for (final String section : prefer.split(";\\s*")) {
+                final String[] parts = section.split("=");
+                if (parts.length == 2) {
```

Comment:
There are plenty of legitimate cases in which `parts.length` will equal 1, but none of those are relevant for Fedora and (AFAICT) LDP in general.